Backend/Remora.Discord.API.Abstractions/API/Rest/IDiscordRestAutoModerationAPI.cs

```diff
@@ -0,0 +1,134 @@
+//
+//  IDiscordRestAutoModerationAPI.cs
+//
+//  Author:
+//       Jarl Gullberg <jarl.gullberg@gmail.com>
+//
+//  Copyright (c) 2017 Jarl Gullberg
+//
+//  This program is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU Lesser General Public License for more details.
+//
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Remora.Discord.API.Abstractions.Objects;
+using Remora.Rest.Core;
+using Remora.Results;
+
+namespace Remora.Discord.API.Abstractions.Rest;
+
+/// <summary>
+/// Represents the Discord Auto Moderation API.
+/// </summary>
+public interface IDiscordRestAutoModerationAPI
+{
+    /// <summary>
+    /// Gets a list of all Auto Moderation rules.
+    /// </summary>
+    /// <param name="guildID">The ID of the guild.</param>
+    /// <param name="ct">The cancellation token for this operation.</param>
+    /// <returns>A retrieval result which may or may not have succeeded.</returns>
+    Task<Result<IReadOnlyList<IAutoModerationRule>>> ListGuildAutoModerationRulesAsync
+    (
+        Snowflake guildID,
+        CancellationToken ct = default
+    );
+
+    /// <summary>
+    /// Gets a single Auto Moderation rule by ID.
+    /// </summary>
+    /// <param name="guildID">The ID of the guild.</param>
+    /// <param name="ruleID">The ID of the rule.</param>
+    /// <param name="ct">The cancellation token for this operation.</param>
+    /// <returns>A retrieval result which may or may not have succeeded.</returns>
+    Task<Result<IAutoModerationRule>> GetGuildAutoModerationRuleAsync
```

Comment:
Guild is not part of the method name. It is documented as GetAutoModerationRule

Reply:
Auto Moderation rules can only apply to a Guild anyway. I don't understand why the docs only append "for Guild" on the list rules endpoint. That said, as you pointed out, the majority of them don't include "Guild", so for the sake of consistency I'll remove it from all the names